pg_search/src/postgres/delete.rs

```diff
@@ -92,23 +113,14 @@ pub extern "C" fn ambulkdelete(
         );
         pg_sys::LockBufferForCleanup(cleanup_buffer);
 
-        // Don't merge here, amvacuumcleanup will merge
-        writer
-            .commit(false)
-            .expect("ambulkdelete: commit should succeed");
+        let bulk_del_data = PgMemoryContexts::CurrentMemoryContext.switch_to(|_context| {
+            let mut opaque = PgBox::<BulkDeleteData>::alloc0();
+            opaque.stats = *stats;
+            opaque.cleanup_lock = cleanup_buffer;
+            opaque.into_pg()
+        });
 
-        pg_sys::UnlockReleaseBuffer(cleanup_buffer);
+        // TODO: Update stats
+        &mut (*bulk_del_data).stats
```

Comment:
can't you just return `bulk_del_data as *mut pg_sys::IndexBulkDeleteResult`?  In fact, don't you need to so that `amvacuumcleanup` gets the correct pointer?